gprofiler/metadata/system_metadata.py

```diff
@@ -237,3 +238,8 @@ def get_infos():
     run_in_ns(["mnt", "uts", "net"], get_infos)
 
     return hostname, distribution, libc_version, mac_address, local_ip
+
+
+@lru_cache(maxsize=None)
+def get_arch() -> str:
+    return platform.uname().machine
```

Comment:
Not sure I like how this value is fetched in 2 different places (in `get_static_system_info` as well). If we ever want to change how this value is fetched (like perhaps exiting the container namespace) we would need to remember to update this as well

Reply:
I already saw it and I don't like it being fetched twice, yeah, but...

>  If we ever want to change how this value is fetched

specifically for this value - we won't ever want - it's the hardware and that's not to be changed (unless someone uses emulators in production, which is stupid). So that's fine IMO.